cmd/pitr/recoverer/recoverer_test.go

```diff
@@ -2,70 +2,42 @@ package recoverer
 
 import (
 	"bytes"
-	"io"
-	"strings"
+	"testing"
 )
 
-type testStorage struct {
-}
-
-type reader struct {
-	r io.Reader
-}
-
-func (r *reader) Read(p []byte) (int, error) {
-	return r.r.Read(p)
-}
-
-func (t *testStorage) GetObject(name string) (io.Reader, error) {
-	if strings.Contains(name, "gtid-set") {
-		buf := bytes.NewBufferString("some-set:1-15")
-		return buf, nil
+func TestGetBucketAndPrefix(t *testing.T) {
```

Comment:
use table tests:
```go
func TestGetBucketAndPrefix(t *testing.T) {
	cases := []string{
		"operator-testing/test",
		"s3://operator-testing/test",
		"https://somedomain/operator-testing/test",
	}

	for _, c := range cases {
		t.Run(c, func(t *testing.T) {
			bucket, prefix, err = getBucketAndPrefix(c)
			if err != nil {
				t.Error("get from 'operator-testing/test'", err.Error())
			}
			if bucket != "operator-testing" && prefix != "test" {
				t.Error("wrong parsing of 'operator-testing/test'")
			}
		}
	})
}
```

and add more cases like
- "operator-testing/test/"
- "operator-testing/test/pitr"
- "https://somedomain/operator-testing"
- "operator-testing"
etc